ls1intum/Apollon
# Add v3 docs (and convenience scripts for working with docs)

### Checklist
- [x] ~~I documented the TypeScript code using JSDoc style~~ (NA).

### Motivation and Context

The documentation is outdated and needs updating. Specifically, the new V3 schema, and the new helper functions for working with Apollon UML models in a safe and backwards compatible manner, should be documented.

### Description

This PR adds the following:

- Documentation for new Apollon v3 schemas and helper functions.
- New scripts in `package.json` for building docs and previewing them locally with auto-rebuild on change.
- A fixes for a minro dependency issue in docs: the docs require `Jinja < 3.1`, but a higher version was being installed without explicitly specifying the version constraint (see [this issue](https://github.com/readthedocs/readthedocs.org/issues/9038)).
- Instructions on README about how to build and work with the docs.

### Steps for Testing

1. `npm run prepare`
2. `npm run docs:prepare`
3. `npm run docs:build` to build the docs in `docs/build/html` \
  -- or -- \
  `npm run docs:watch`, to preview the docs on `localhost:8088`

The following requirements should be met:

- The scripts should work and should be explained properly by the README
- `Apollon API` entry on the docs (built or previewed) should be updated to reflect model 3 schema
- A new `UML Model Helpers` entry on the docs (built or previewed) should be found, detailing the helper functions with sufficient clarity.


### Screenshots

<img width="1096" alt="Screenshot 2023-11-27 at 15 35 32" src="https://github.com/ls1intum/Apollon/assets/13572283/a56422c2-e8d5-432e-9035-e5441264ce40">
